internal/appsec/remoteconfig.go

```diff
@@ -228,17 +293,25 @@ func (a *appsec) enableRemoteActivation() error {
 	}
 	a.registerRCProduct(rc.ProductASMFeatures)
 	a.registerRCCapability(remoteconfig.ASMActivation)
-	a.registerRCCallback(a.asmFeaturesCallback, rc.ProductASMFeatures)
```

Comment:
It would be actually cleaner to keep this into a separate callback so that this is called only when the capability is set. Whereas we currently always try to see if `ASM_FEATURES` is present in the global umbrella callback.

Reply:
I think it would actually be more costly than a simple `if` check in the umbrella callback. This means an entirely new function call along with a new config status merge, separate from the main umbrella callback one. Also this kinda takes away from the idea to have a single callback that handles all updates. Are you sure about this?

Reply:
Ok, let's keep it that way and we see how it evolves.

> Also this kinda takes away from the idea to have a single callback that handles all updates.

Single callback for everything interdependent indeed, such as all the RC products required for the appsec rules. Otherwise, I still think splitting callbacks per "business" can be worth it.